packages/product-components/src/components/SelectAssetModal/SelectAssetModal.tsx

```diff
@@ -1,137 +1,184 @@
-import { useEffect, useState } from 'react';
+import { useMemo, useState } from 'react';
 import {
-    AssetLogo,
-    Button,
     Column,
     Icon,
     Input,
     NewModal,
-    Row,
-    useElevation,
     useScrollShadow,
     VirtualizedList,
-    Text,
 } from '@trezor/components';
 import { mapElevationToBackgroundToken, spacings } from '@trezor/theme';
 import { AssetItem } from './AssetItem';
+import { NetworkTabs } from './NetworkTabs';
+import { useIntl } from 'react-intl';
+import { AssetItemNotFound } from './AssetItemNotFound';
 
-const LIST_HEIGHT = '50vh';
-const height = 60;
+export interface SelectAssetOptionCurrencyProps {
+    type: 'currency';
+    value: string; // CryptoId (networkId + contractAddress)
+    label: string; // token shortcut
+    cryptoName: string | undefined; // full name
+    coingeckoId: string; // CryptoId (networkId)
+    contractAddress?: string; // CryptoId (contractAddress)
+    networkName?: string;
+}
+export interface SelectAssetOptionGroupProps {
+    type: 'group';
+    label: string;
+    networkName?: string;
+}
+export type SelectAssetOptionProps = SelectAssetOptionCurrencyProps | SelectAssetOptionGroupProps;
 
-type Data = {
-    id: number;
+export interface SelectAssetNetworkProps {
+    name: string;
+    symbol: string;
+    coingeckoId: string;
+    coingeckoNativeId?: string;
+}
+
+export type SelectAssetSearchCategoryType = {
+    coingeckoId: string;
+    coingeckoNativeId?: string;
+} | null;
+
+export interface SelectAssetModalProps {
+    options: SelectAssetOptionProps[];
+    networkCategories: SelectAssetNetworkProps[];
+    onSelectAssetModal: (selectedAsset: string) => void;
+    onFavoriteClick?: (isFavorite: boolean) => void;
+    onClose: () => void;
+}
+
+interface AssetProps
+    extends Pick<SelectAssetOptionCurrencyProps, 'coingeckoId' | 'contractAddress'> {
+    cryptoId: string;
     symbol: string;
     name: string;
     height: number;
     isFavorite: boolean;
-    badge: string;
-};
-
-const getData = (end: number, tokens: Array<string>): Array<Data> =>
-    Array.from({ length: end }, (_, i) => ({
-        id: i,
-        symbol: `symbol ${i}`,
-        name: `name ${i}`,
-        badge: tokens[Math.floor(Math.random() * tokens.length)],
-        isFavorite: Math.random() < 0.1,
-        height,
-    }));
+    badge: string | undefined;
+}
 
-export type SelectAssetModalProps = {
-    onSelectAssetModal: (selectedAsset: string) => void;
-    onFavoriteClick: (isFavorite: boolean) => void;
-};
+const LIST_HEIGHT = 'calc(80vh - 267px)'; // 267px is height header part of modal
+const height = 60;
```

Comment:
I'm not sure what does this mean. Maybe it's item height?

In that case I suggest:
```
const ITEM_HEIGHT = 60